src/main/scala/org/hammerlab/guacamole/commands/StructuralVariantCaller.scala

```diff
@@ -180,6 +182,99 @@ object StructuralVariant {
       graph
     }
 
+    case class SVClique(
+      reads: Set[PairedMappedRead],
+      span: GenomeRange,
+      wiggle: Long
+    )
+
+    def findOneClique(g: PairedReadGraph, maxNormalInsertSize: Int): SVClique = {
+      // Seed the clique with the source of the lowest-weight edge.
+      val edges = g.edges.toList.sortBy(_.weight)
+      val bestEdge = edges.head
+
+      val initPair = bestEdge.nodes.minBy(_.minPos).value
+      var (pairMin, svStart, svEnd, _) = initPair.startsAndStops
+
+      // this is the "wiggle room": by how much can the SV shrink and still be compatible with all reads in the clique?
+      var wiggle = initPair.insertSize - maxNormalInsertSize - (svEnd - svStart)
+
+      val clique = mutable.Set(initPair)
+
+      val graphSize = g.nodes.length
+
+      var cliqueGrow = true
+      while (cliqueGrow && clique.size < graphSize) {
+        cliqueGrow = false
+        // Try to find a node which:
+        // 1. isn't currently in the clique
+        // 2. is connected to every node in the clique
+        // 3. doesn't perturb the SV in a way that makes it incompatible with any reads in the clique
+        mapFirst(edges, (edge: PairedReadGraph#EdgeT) => {
+          val nodes = edge.nodes.toSeq
+          val Seq(node1, node2) = nodes
+          val node1InClique = clique.contains(node1.value)
+          val node2InClique = clique.contains(node2.value)
+
+          // The edge should have one end in the clique and one end out of it.
+          val node = (node1InClique, node2InClique) match {
+            case (false, true) => Some(node1)
+            case (true, false) => Some(node2)
+            case _             => None
+          }
+
+          // TODO: DELLY caches a list of incompatible vertices
+
+          // The new node should be fully connected with the rest of the clique.
+          node filter { n =>
+            {
+              clique.subsetOf(n.neighbors map { _.value })
+            }
+          } filter { n =>
+            // And it should be compatible with the existing SV.
+            {
+              val (_, nodeGapMin, nodeGapMax, _) = n.value.startsAndStops
+              val newSvStart = Math.max(svStart, nodeGapMin)
+              val newSvEnd = Math.min(svEnd, nodeGapMax)
+
+              // The amount of wiggle room could shrink either because of the new read, or because it shrinks the
+              // the induced structural variant and pulls another read closer to being incompatible.
+              val wiggleNewRead = n.value.insertSize - maxNormalInsertSize - (newSvEnd - newSvStart)
+              val wiggleChange = wiggle + (svEnd - svStart) - (newSvEnd - newSvStart)
+              val newWiggle = Math.max(wiggleNewRead, wiggleChange)
+
+              if ((newSvStart < newSvEnd) && (newWiggle <= 0)) {
+                svStart = newSvStart
+                svEnd = newSvEnd
+                wiggle = newWiggle
+                cliqueGrow = true
```

Comment:
you could kill `cliqueGrow` if you just check `.isDefined` on the `mapFirst`'s returned `Option`, I think?